src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -34,39 +34,49 @@ ov::pass::MarkDequantizationSubgraph::MarkDequantizationSubgraph(const element::
 
     ov::matcher_pass_callback callback = [=](pattern::Matcher& m) -> bool {
         const auto& pattern_map = m.get_pattern_value_map();
-        const auto& convert = pattern_map.at(convert_pattern).get_node_shared_ptr();
-        const auto& input_precision = convert->get_input_element_type(0);
+        auto convert = pattern_map.at(convert_pattern).get_node_shared_ptr();
+        auto input = pattern_map.at(input_pattern).get_node_shared_ptr();
+        const auto multiply = m.get_match_root();
 
-        // validation by Convert operation input precisions
-        if (!precisions.empty()) {
-            if (std::find(precisions.begin(), precisions.end(), input_precision) == precisions.end()) {
-                return false;
+        auto subtract_it = pattern_map.find(subtract_pattern);
+        if (subtract_it == pattern_map.end()) {
+            for (size_t i = 0; i < multiply->get_input_size(); i++) {
+                const auto node = ov::as_type_ptr<opset10::Convert>(multiply->get_input_node_shared_ptr(i));
+                if (node && std::find(precisions.begin(), precisions.end(), node->get_input_element_type(0)) !=
+                                precisions.end()) {
+                    convert = node;
+                    input = convert->get_input_node_shared_ptr(0);
+                }
             }
         }
 
-        const auto& input = pattern_map.at(input_pattern);
+        // validation by Convert operation input precisions
+        const auto& input_precision = input->get_output_element_type(0);
+        if (std::find(precisions.begin(), precisions.end(), input_precision) == precisions.end()) {
+            return false;
+        }
```

Comment:
1. Can `zero_point_pattern` be a `Constant` operation pattern?

- If yes, then can you simplify the logic for constant values?
- If no, then can you generalize the code to find input operation on data path for the `Multiply` and the `Subtract`. Please, see notes below for details.

Minor comments:

- If `Subtract` is absent, then precision validation logic works twice. It will be fixed if you simplify the logic.
- Question. If `Subtract` is absent, then both `Multuply` parent nodes are checked and anyone can be used as data path. For `Subtract`, only the first can be used as data path. Is it expected? It will be fixed if you simplify the logic.
 
Notes:

- we should keep in mind that dequantization pattern works with element-wise operations with constant values. But the transformation is executed before common optimization transformations and the constant path can be appeared after them.

- `input` variable value is pointer of `Convert` parent on data path. But if `Subtract` is absent then `input` can be `Muliply` constant (if `Constant` has integer values). It doesn't happened in real model for `Multiply`, but in case of generalization you should think about that for `Subtract`.